### PR TITLE
Added scale tests tooling and workloads

### PR DESCRIPTION
**Pull request by akirillov**
### What changes were proposed in this pull request?
Resolves [DCOS-59718: Migrate deployment scripts to KUDO Spark for MWT](https://jira.mesosphere.com/browse/DCOS-59718)

This PR adds scripts for:
- per-namespace multi-instance deployment of KUDO Spark Operator
- gang submission of multiple templated Spark Application for running the workload
- running dataset generator for sorting benchmark

Dashboards are migrated from [ds-tooling/phoenix-mwt/spark-operator](https://github.com/mesosphere/data-services-tooling/tree/master/phoenix-mwt/spark-operator)

### Why are the changes needed?
These changes were required to address the scale-tests requirements for KUDO Spark Operator:
- multi-instance deployment
- multi-namespace workload submission
- metrics
- automation

### How were the changes tested?
By using developed scripts and dashboards during MWT

### CODE REVIEW

File: scale-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+Scale Tests Tooling
+---
+
+## Before you start
+
+* `kubectl` should be configured and point to the cluster under test
+* KUDO CLI initialized with `kubect kudo init [--client-only]`
```
Comment (alembiewski) [inline scale-tests/README.md]:
```suggestion
* KUDO CLI initialized with `kubectl kudo init [--client-only]`
```

File: scale-tests/scripts/run.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  echo "Usage:" >&2
+  echo "  $0 <number of applications to create> <number of namespaces to submit apps to>" >&2
+  exit 1
+fi
+
+. ${SCRIPT_DIR}/aws_credentials.sh
+
+for n in $(seq ${2}); do
+#    NAMESPACE="${NAMESPACE_PREFIX}-${n}"
+    NAMESPACE=spark
```
Comment (alembiewski) [inline scale-tests/scripts/run.sh]: This line looks like a leftover from local testing

File: scale-tests/templates/sort-application.tmpl [hunk 3]
```diff
@@ -0,0 +1,43 @@
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: sort-dataset-generator
```
Comment (alembiewski) [inline scale-tests/templates/sort-application.tmpl]: Should we rename this to something like `sort-dataset-application` ?

File: scale-tests/templates/sort-application.tmpl [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    "spark.hadoop.fs.s3a.secret.key": "AWS_SECRET_ACCESS_KEY"
+    "spark.hadoop.fs.s3a.session.token": "AWS_SESSION_TOKEN"
+  sparkVersion: "2.4.3"
+  restartPolicy:
+    type: Never
+  driver:
+    cores: 2
+    memory: "2g"
+    labels:
+      version: 2.4.3
```
Comment (alembiewski) [inline scale-tests/templates/sort-application.tmpl]: It looks like we are missing `metrics-exposed: "true"` label here, when I tested locally, Spark metrics weren't reported correctly

File: scale-tests/templates/sort-application.tmpl [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    serviceAccount: spark-driver
+  executor:
+    cores: 1
+    instances: NUM_EXECUTORS
+    memory: "3g"
+    labels:
```
Comment (alembiewski) [inline scale-tests/templates/sort-application.tmpl]: ditto

File: scale-tests/scripts/install.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+for i in $(seq ${1}); do
+    NAMESPACE="${NAMESPACE_PREFIX}-${i}"
+    echo "Creating namespace $NAMESPACE"
+    sed 's|SPARK_NAMESPACE|'"${NAMESPACE}"'|g' ${TEMPLATES_DIR}/namespace.tmpl | kubectl apply -f -
+
+    kubectl kudo --namespace "${NAMESPACE}" install --instance "${INSTANCE_NAME_PREFIX}-${i}" "${OPERATOR_DIR}" -p operatorVersion="${OPERATOR_VERSION}"
+    kubectl apply --namespace "${NAMESPACE}" -f ${SPECS_DIR}/spark-driver-rbac.yaml
+    # Metrics
+    kubectl apply --namespace "${NAMESPACE}" -f ${SPECS_DIR}/spark-operator-service.yaml
```
Comment (alembiewski) [inline scale-tests/scripts/install.sh]: It looks like the file name is different - should it be `spark-application-metrics-service.yaml`?

File: scale-tests/templates/sort-application.tmpl [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      version: 2.4.3
+  monitoring:
+    exposeDriverMetrics: true
+    exposeExecutorMetrics: true
+    prometheus:
+      jmxExporterJar: "/prometheus/jmx_prometheus_javaagent-0.11.0.jar"
+      port: 49151
```
Comment (alembiewski) [inline scale-tests/templates/sort-application.tmpl]: nit: when tested locally, I modified the port value to `8090` (to be the same, as in `spark-application-metrics-service.yaml` spec).

File: scale-tests/scripts/run.sh (hunk 2 above)
Comment (akirillov) [reply scale-tests/scripts/run.sh]: fixed

File: scale-tests/templates/sort-application.tmpl (hunk 4 above)
Comment (akirillov) [reply scale-tests/templates/sort-application.tmpl]: added

File: scale-tests/templates/sort-application.tmpl (hunk 5 above)
Comment (akirillov) [reply scale-tests/templates/sort-application.tmpl]: added

File: scale-tests/scripts/install.sh (hunk 6 above)
Comment (akirillov) [reply scale-tests/scripts/install.sh]: renamed

File: scale-tests/templates/sort-application.tmpl (hunk 7 above)
Comment (akirillov) [reply scale-tests/templates/sort-application.tmpl]: updated,  thanks for catching this

**Pull request merged**